libraries/validate/src/index.js

```diff
@@ -118,10 +118,12 @@ class SchemaSet {
     return (obj, id) => {
       id = id.replace(/#$/, '');
       id = id.replace(/\.ya?ml$/, '.json');
-      if (!_.endsWith(id, '.json')) {
-        id += '.json';
+      if (!/#/.test(id)) {
+        if (!_.endsWith(id, '.json')) {
+          id += '.json';
+        }
+        id += '#';
```

Comment:
I'm surprised we've had this wrong for so long. This was added like 4.5 years ago!

Reply:
I think this is really just adding a feature we've never needed before -- all of the other validation we do is by "file", not sub-schema.